### ISSUE TITLE
Blazor script security problems RC1

### ISSUE THREAD

**Issue by damienbod** [dotnet/aspnetcore#36529]
I updated a Blazor app to the latest RC1 version and now my build has security warnings due to the generated JavaScript used by Blazor. The **dotnet.6.0.0-rc.1.21451.13.js** warning is:  **Use of externally-controlled format string**

![image](https://user-images.githubusercontent.com/3442158/133384411-0ba88052-f576-45ca-9809-6f0b55b45ba9.png)

It is also not possible to create a good CSP due to the generated JS used by Blazor. This leaves Blazor deployed apps open to attacks especially if the app is secured using tokens and storing these in local storage like AAD does.

Would it be possible to fix this before .NET 6 release. I really like Blazor and want to use this in productive deployments but this is a real blocker for me.

Greetings Damien

**Comment by SteveSandersonMS** [comment]
@damienbod This definitely sounds like something we'd want to look into. There's a strong chance this is a false positive, but I don't know how the code scanning works, or what exactly it's being triggered by, so I can't know for sure.

Can you give us some suggestions on how to repro this? Is your pipeline public somewhere, or could you let us know what's needed to get the same issue on a repo and pipeline that we set up?

Even before that it would be super useful if you could post any other info that the warning gives you (the "show more details" link, and whatever "Show paths" does).

**Comment by damienbod** [comment]
Hi @SteveSandersonMS thanks for answering.

This repo, I can add you as an admin if you want. The PR fails with the RC1 update.

https://github.com/damienbod/AspNetCore6Experiments

I use CodeQL here:

https://github.com/damienbod/AspNetCore6Experiments/blob/main/.github/workflows/codeql-analysis.yml

Greetings Damien

**Comment by SteveSandersonMS** [comment]
@damienbod I investigated and wrote up a report at https://github.com/dotnet/runtime/issues/59147

In summary, I do not see how this could be a real issue, but I appreciate that it's inconvenient that you see the report from codeql. Codeql's JavaScript analysis may not be aware that some JS code runs on the server, and some on the client, and the security characteristics of the two are very different. If we could adapt the code to avoid falsely triggering, that would be great, hence the issue I filed on the runtime.

I'll close this as external and superseded by https://github.com/dotnet/runtime/issues/59147
